kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -313,6 +313,14 @@ public void definitionChecks(Set<Module> modules) {
 
     // Extra checks just for the prover specification.
     public Module proverChecks(Module specModule, Module mainDefModule) {
+        // check rogue syntax in spec module
+        Set<Sentence> toCheck = mutable(specModule.sentences().$minus$minus(mainDefModule.sentences()));
+        for (Sentence s : toCheck)
+            if (s instanceof Production || s instanceof SyntaxSort || s instanceof SyntaxLexical ||
+                    s instanceof SyntaxPriority || s instanceof SyntaxAssociativity)
```

Comment:
There isn't some `Syntax` super-class you can use instead of listing all these out?

@dwightguth should probably check that this check is being done in the correct place, output looks OK to me.

Reply:
```
case class Production(klabel: Option[KLabel], params: Seq[Sort], sort: Sort, items: Seq[ProductionItem], att: Att)
  extends Sentence with ProductionToString
```
They all extend `Sentence` directly, I had to list them all.

Reply:
Would this message be better?
`Found syntax declaration in proof module. This can cause parsing issues since it's not visible from the main module.`

Reply:
rather than using a `instanceof`, you ought to be able to say `sentence.isSyntax()`